internal/clierr/api_error.go

```diff
@@ -0,0 +1,116 @@
+package clierr
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/armory/armory-cli/internal/clierr/exitcodes"
+	"github.com/armory/armory-cli/pkg/console"
+	"github.com/fatih/color"
+)
+
+type (
+	// ApiErrorResponse the error wrapper that is returned from API errors with Armory APIs.
+	ApiErrorResponse struct {
+		// ErrorID this is a guid that is generated for each API error and can be used to look up the error in the log aggregator.
+		ErrorID string `json:"error_id"`
+		// Errors more often than not this is a single error, the notable exception is when request objects fail validation, in that case, there is an error for each validation violation.
+		Errors []ApiErrorDTO `json:"errors"`
+	}
+
+	// ApiErrorDTO the struct that houses the actual error message.
+	ApiErrorDTO struct {
+		// Message the human-readable error message.
+		Message string `json:"message"`
+		// Metadata optionally used to add context around an error, such as JSON deserialization errors.
+		Metadata map[string]any `json:"metadata,omitempty"`
+		Code     any            `json:"code"`
+	}
+
+	// APIError an error that can be returned from a cobra command that wraps an API error from the Armory API.
+	APIError struct {
+		message          string
+		httpStatusCode   int
+		apiErrorResponse *ApiErrorResponse
+		exitCode         exitcodes.ExitCode
+	}
+)
+
+// Error the string representing the error.
+func (a *APIError) Error() string {
+	return fmt.Sprintf("an API error occurred, msg: %v, status code: %v", a.message, a.httpStatusCode)
+}
+
+// DetailedError returns a human readable multiline colored error message.
+func (a *APIError) DetailedError() string {
+	msg := color.New(color.FgRed, color.Bold).Sprintln(a.message)
```

Comment:
i'm curious - would it work with various output format specifiers? 

what if i want to have detailed output information, formatted as yaml or json?

Reply:
I replied to this here: https://github.com/armory-io/armory-cli/pull/171#discussion_r1270795159